hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/HBaseRpcControllerImpl.java

```diff
@@ -281,4 +282,15 @@ public String toString() {
       + exception + ", regionInfo=" + regionInfo + ", priority=" + priority + ", cellScanner="
       + cellScanner + '}';
   }
+
+  public static void configureRequestAttributes(RpcController rpcController,
+    Map<String, byte[]> requestAttributes) {
+    if (
+      !requestAttributes.isEmpty() && rpcController != null
+        && rpcController instanceof HBaseRpcController
+    ) {
+      HBaseRpcController controller = (HBaseRpcController) rpcController;
+      controller.setRequestAttributes(requestAttributes);
+    }
+  }
```

Comment:
I think we could move this to the RpcControllerFactory; I originally put this here because I thought we'd need to configure request attributes in several places before finding the AsyncProcess class